### PR TITLE
add logic to custom cluster roles check

### PR DESCRIPTION
**Pull request by slickwarren**
## Issue: 

quick fix for last commit
 
## Solution

check for each role in list instead of -> string
 
### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

**Review by markusewalker — APPROVED** [review]

**Review by markusewalker — DISMISSED** [review]

File: tests/v2/validation/rbac/rbac.go [hunk 1]
```diff
@@ -46,7 +48,7 @@ const (
 )
 
 type ClusterConfig struct {
-	nodeRoles            []string
+	NodesAndRoles        []nodepools.NodeRoles `json:"nodesAndRolesRKE1" yaml:"nodesAndRolesRKE1" default:"[]"`
```
Comment (igomez06) [inline tests/v2/validation/rbac/rbac.go:51]: can you keep this private? Otherwise this PR is good to go!

Comment (slickwarren) [reply tests/v2/validation/rbac/rbac.go]: done

**Review by markusewalker — DISMISSED** [review]

**Review by igomez06 — APPROVED** [review]

File: tests/v2/validation/rbac/rbac.go [hunk 2]
```diff
@@ -46,7 +48,7 @@ const (
 )
 
 type ClusterConfig struct {
-	nodeRoles            []string
+	nodesAndRoles        []nodepools.NodeRoles `json:"nodesAndRolesRKE1" yaml:"nodesAndRolesRKE1" default:"[]"`
```
Comment (igomez06) [inline tests/v2/validation/rbac/rbac.go:51]: actually one last thing do we need the json tags here?

Comment (slickwarren) [reply tests/v2/validation/rbac/rbac.go]: I guess not 🥲

**Review by igomez06 — APPROVED** [review]

**Review by markusewalker — APPROVED** [review]

**Pull request merged**